lib/libmuser.c

```diff
@@ -170,7 +170,8 @@ irqs_set_data_none(lm_ctx_t *lm_ctx, struct vfio_irq_set *irq_set)
     long ret;
     eventfd_t val;
 
-    for (i = irq_set->start; i < irq_set->start + irq_set->count; i++) {
+    for (i = (int)irq_set->start; i < (int)(irq_set->start + irq_set->count);
```

Comment:
Instead of typecasting it everywhere, you can just declare 'i' as unsigned int ?

Reply:
It's better to use signed because overflow of a signed int is defined while overflow of unsigned is not.

Reply:
irq_set->start and irq_set->count are __u32(unsigned int). So if 'i' is (int) in reality it cannot accommodate all the values of __u32 it will wrap-around.
Its other way the overflow of unsigned is defined,
https://www.gnu.org/software/autoconf/manual/autoconf-2.63/html_node/Integer-Overflow-Basics.html

Reply:
you're right, I'll fix it.

Reply:
fixed